internal/consensus/state.go

```diff
@@ -1265,8 +1265,9 @@ func (cs *State) createProposalBlock() (block *types.Block, blockParts *types.Pa
 
 // Enter: `timeoutPropose` after entering Propose.
 // Enter: proposal block and POL is ready.
-// Prevote for LockedBlock if we're locked, or ProposalBlock if valid.
-// Otherwise vote nil.
+// If we received a valid proposal within this round that matches the block we are
+// locked on or matches a block that received a POL in a previous round, prevote
+// for the proposal, otherwise vote nil.
```

Comment:
```suggestion
// If we received a valid proposal within this round that matches one of:
// * the block we are locked on
// * a block that received a POL in a previous round
// then prevote for the proposal. Otherwise vote nil.
```

Minor nit that helped readability for me